src/vs/platform/telemetry/node/1dsAppender.ts

```diff
@@ -0,0 +1,155 @@
+					headers: {
+						...payload.headers,
+						'Content-Type': 'application/json',
+						'Content-Length': Buffer.byteLength(payload.data)
+					}
+				};
+				const req = https.request(payload.urlString, options, res => {
+					res.on('data', function (responseData) {
+						oncomplete(res.statusCode ?? 200, res.headers as Record<string, any>, responseData.toString());
+					});
+				});
+				req.write(payload.data);
+				req.end();
+			}
+		};
+		// Configure the channel to use a XHR Request override since it's not available in node
+		const channelConfig: IChannelConfiguration = {
+			alwaysUseXhrOverride: true,
+			httpXHROverride: customHttpXHROverride
+		};
+		coreConfig.extensionConfig[collectorChannelPlugin.identifier] = channelConfig;
+	}
+
+	appInsightsCore.initialize(coreConfig, []);
+
+	appInsightsCore.addTelemetryInitializer((envelope) => {
+		if (envelope.tags) {
+			// Sets it to be internal only
+			envelope.tags['utc.flags'] = 880971612160;
+		}
+	});
+
+	return appInsightsCore;
+}
+
+
+export class OneDsAppender implements ITelemetryAppender {
+
+	private _aiCore: AppInsightsCore | undefined;
+	private _iKey: string | undefined;
+	private _asyncAiCore: Promise<AppInsightsCore> | null;
+
+	constructor(
+		private _eventPrefix: string,
+		private _defaultData: { [key: string]: any } | null,
+		iKeyOrClientFactory: string | (() => AppInsightsCore), // allow factory function for testing
+	) {
+		if (!this._defaultData) {
+			this._defaultData = Object.create(null);
+		}
+
+		if (typeof iKeyOrClientFactory === 'function') {
+			this._aiCore = iKeyOrClientFactory();
+		} else {
+			this._iKey = iKeyOrClientFactory;
+		}
+		this._asyncAiCore = null;
+	}
+
+	private _withAIClient(callback: (aiCore: AppInsightsCore) => void): void {
+		if (!this._aiCore) {
+			return;
+		}
+
+		if (this._aiCore) {
+			callback(this._aiCore);
+			return;
+		}
+
+		if (this._iKey && !this._asyncAiCore) {
+			this._asyncAiCore = getClient(this._iKey);
+			this._iKey = undefined;
+		}
+
+		this._asyncAiCore?.then(
+			(aiClient) => {
+				callback(aiClient);
+			},
+			(err) => {
+				onUnexpectedError(err);
+				console.error(err);
+			}
+		);
+	}
+
+	log(eventName: string, data?: any): void {
+		if (!this._aiCore) {
+			return;
+		}
+		data = mixin(data, this._defaultData);
+		data = validateTelemetryData(data);
+
+		// Attemps to suppress https://github.com/microsoft/vscode/issues/140624
+		try {
+			this._withAIClient((aiClient) => aiClient.track({
+				name: this._eventPrefix + '/' + eventName,
+				data: { ...data.properties, ...data.measurements },
+
+			}));
+		} catch { }
+	}
+
+	flush(): Promise<any> {
+		if (this._aiCore) {
+			return new Promise(resolve => {
+				this._withAIClient((aiClient) => {
+					// Attempts to suppress https://github.com/microsoft/vscode/issues/140624
+					try {
+						aiClient.flush(true,
+							() => {
+								// all data flushed
+								this._aiCore?.unload(true, () => this._aiCore = undefined);
```

Comment:
(nice!) Note: calling unload will also cause events to be flushed, so technically you don't need to call flush first (when unload is present)

Reply:
Ok thanks I will switch to calling just unload